features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/OtelTracerProvider.kt

```diff
@@ -228,11 +241,59 @@ class OtelTracerProvider(
         // endregion
     }
 
+    // region Internal
+
+    private fun resolveInstrumentationScopeName(instrumentationScopeName: String): String {
+        return if (instrumentationScopeName.trim { it <= ' ' }.isEmpty()) {
+            DEFAULT_TRACER_NAME
+        } else {
+            instrumentationScopeName
+        }
+    }
+
+    private fun resolveSpanBuilderDecorator(): (SpanBuilder) -> SpanBuilder {
+        return if (bundleWithRumEnabled) {
+            resolveSpanBuilderDecoratorFromContext()
+        } else {
+            NO_OP_SPAN_BUILDER_DECORATOR
+        }
+    }
+
+    private fun resolveSpanBuilderDecoratorFromContext(): (SpanBuilder) -> SpanBuilder = { spanBuilder ->
```

Comment:
minor: I think the name can be improved here (can be done in another PR) - nothing in the name points that it will be RUM content applied. Maybe something like `rumAwareSpanBuilderDecorator` for this method will be better (also this can be just a property, not a method).

Reply:
yes but that is the idea, this time is for RUM but it should be feature agnostic that's the reason why it is not having any RUM mentioned.

Reply:
then still the name doesn't resolve anything `from` context, it decorates `with` context.